core/trino-main/src/main/java/io/trino/sql/planner/FragmentTableScanCounter.java

```diff
@@ -43,12 +47,35 @@ public static boolean hasMultipleSources(PlanNode... nodes)
     {
         int count = 0;
         for (PlanNode node : nodes) {
-            count += node.accept(new Visitor(), null);
+            count += node.accept(new TableScanCounterVisitor(), null);
         }
         return count > 1;
     }
 
-    private static class Visitor
```

Comment:
you probably don't needs visitor anymore, because we have `io.trino.sql.planner.optimizations.PlanNodeSearcher` and this class predates it and you could use it